test/operations.py

```diff
@@ -0,0 +1,61 @@
+import requests
+import json
+from typing import Optional, Tuple
+
+
+def create_index(endpoint: str, index_name: str, auth: Optional[Tuple[str, str]] = None, data: Optional[dict] = None):
+    response = requests.put(f'{endpoint}/{index_name}', auth=auth, verify=False)
+
+    return response
+
+
+def check_index(endpoint: str, index_name: str, auth: Optional[Tuple[str, str]] = None, data: Optional[dict] = None):
+    response = requests.get(f'{endpoint}/{index_name}', auth=auth, verify=False)
+
+    return response
+
+
+def delete_index(endpoint: str, index_name: str, auth: Optional[Tuple[str, str]] = None, data: Optional[dict] = None):
+    response = requests.delete(f'{endpoint}/{index_name}', auth=auth, verify=False)
+
+    return response
+
+
+def delete_document(endpoint: str, index_name: str, doc_id: str, auth: Optional[Tuple[str, str]] = None):
+    response = requests.delete(f'{endpoint}/{index_name}/_doc/{doc_id}', auth=auth, verify=False)
+
+    return response
+
+
+def create_document(endpoint: str, index_name: str, doc_id: str, auth: Optional[Tuple[str, str]] = None,
+                    data: Optional[dict] = None):
+    document = {
+        'title': 'Test Document',
+        'content': 'This is a sample document for testing OpenSearch.'
+    }
+    url = f'{endpoint}/{index_name}/_doc/{doc_id}'
+    headers = {'Content-Type': 'application/json'}
+
+    response = requests.put(url, headers=headers, data=json.dumps(document), auth=auth, verify=False)
+
+    return response
+
+
+def check_document(endpoint, index_name, doc_id, auth=None):
+    url = f'{endpoint}/{index_name}/_doc/{doc_id}'
+    headers = {'Content-Type': 'application/json'}
+
+    response = requests.get(url, headers=headers, auth=auth, verify=False)
+
+    return response
+
+
+def get_document(endpoint: str, index_name: str, doc_id: str, auth: Optional[Tuple[str, str]] = None):
+    url = f'{endpoint}/{index_name}/_doc/{doc_id}'
+    headers = {'Content-Type': 'application/json'}
+
+    response = requests.get(url, headers=headers, auth=auth, verify=False)
+    document = response.json()
+    content = document['_source']
+
+    return content
```

Comment:
Okay, I just read this function it does seem to be the "retrieve_document" I described above. In that case, I'm not sure what the purpose of `check_document` is. These two functions call the same API, right? The second just returns the original document?

I wonder if they should be combined into one function: return the document, or return None if it doesn't exist. Not sure if that's the right solution, but confused that they both exist.